tests/smoke/events.js

```diff
@@ -47,7 +47,46 @@
 
 				wait();
 			});
+		},
+		'test future spellchecks only check the current element': function() {
+			var bot = this.editorBot,
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter,
+				observer = observeSpellCheckEvents(editor),
+				starterHtml = '<p>asdf jkl dzxda</p><p>asdf jkl^</p>';
+
+			bot.setHtmlWithSelection(starterHtml);
 
+			resumeAfter(editor, 'spellCheckComplete', function () {
+				// first run
+				observer.assert(["spellCheckComplete", "startMarkTypos", "startCheckWordsAjax", "startScanWords"]);
+
+				// make a new observer to clear the events
+
+				observer = observeSpellCheckEvents(editor);
+
+				resumeAfter(editor, 'spellCheckComplete', function () {
+					var secondParagraph = editor.editable().getChild(1);
```

Comment:
this kinda sucks to read because you have to register this callback before you actually do the thing that triggers the events...hm